sirepo/auth/ldap.py

```diff
@@ -0,0 +1,44 @@
+"""LDAP login
+:copyright: Copyright (c) 2018-2022 RadiaSoft LLC.  All Rights Reserved.
+:license: http://www.apache.org/licenses/LICENSE-2.0.html
+"""
+from pykern.pkdebug import pkdlog, pkdp
+from pykern.pkcollections import PKDict
+import ldap
+import sirepo.quest
+
+AUTH_METHOD = "ldap"
+
+#: User can see it
+AUTH_METHOD_VISIBLE = True
+
+
+class API(sirepo.quest.API):
+    @sirepo.quest.Spec("require_cookie_sentinel")
+    def api_authLdapAuthorized(self):
+        """Python-LDAP client authentication
+        Bind to server and authenticate
+        """
+        # Test User Credentials:
+        # username: Archimedes of Syracuse
+        # password: eureka
+
+        LDAP_SERVER = "ldap://10.10.10.10:389"
+        BASE_DN = "dc=example,dc=com"
```

Comment:
This must be configured with the default for dev being example and com. Did you look at the customer's example app? It may effect the object to search which is likely "user".